lib/jekyll-readme-index/generator.rb

```diff
@@ -26,15 +28,38 @@ def readme
     end
 
     def index?
-      (site.pages + site.static_files).any? { |file| file.url =~ INDEX_REGEX }
+      relative_index?(readme)
     end
 
     def page
+      relative_page(readme)
+    end
+
+    def readmes
+      site.static_files.each_with_object([]) do |file, memo|
+        memo << file if file.relative_path =~ READMES_REGEX
+      end
+    end
+
+    def relative_index?(readme)
```

Comment:
Instead of passing around the readme itself, what if we had one method that got the relative path of the README's containing folder, and then another method that determined if that folder had an index? That way we can centralized the path traversal logic, and things should read a bit better to humans.